tests/integration/test_charm.py

```diff
@@ -34,3 +41,107 @@ async def test_is_active(ops_test: OpsTest):
         raise_on_error=True,
         timeout=300,
     )
+
+
+def _safe_load_file_to_text(filename: str):
+    """Returns the contents of filename if it is an existing file, else it returns filename"""
+    try:
+        text = Path(filename).read_text()
+    except FileNotFoundError:
+        text = filename
+    return text
+
+
+def create_all_from_yaml(
+    yaml_file: str,
+    lightkube_client: lightkube.Client = None,
+):
+    """Creates all k8s resources listed in a YAML file via lightkube
+    Args:
+        yaml_file (str or Path): Either a string filename or a string of valid YAML.  Will attempt
+                                 to open a filename at this path, failing back to interpreting the
+                                 string directly as YAML.
+        lightkube_client: Instantiated lightkube client or None
+    """
+
+    yaml_text = _safe_load_file_to_text(yaml_file)
+
+    if lightkube_client is None:
+        lightkube_client = lightkube.Client()
+
+    for obj in codecs.load_all_yaml(yaml_text):
+        try:
+            lightkube_client.create(obj)
+        except lightkube.core.exceptions.ApiError as e:
+            raise ValueError(f"unable to create resource from file {yaml_file} becuase of {e}")
+
+
+def delete_all_from_yaml(
```

Comment:
Same comment as above - should be able to use code from [here](https://github.com/canonical/charmed-kubeflow-chisme/blob/main/src/charmed_kubeflow_chisme/lightkube/batch/_many.py)